xlators/cluster/afr/src/afr-open.c

```diff
@@ -235,20 +236,43 @@ afr_openfd_fix_open_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
     call_count = afr_frame_return(frame);
     if (call_count == 0)
         AFR_STACK_DESTROY(frame);
-
     return 0;
 }
 
+static void
+afr_fd_ctx_reset_need_open(fd_t *fd, xlator_t *this, unsigned char *need_open)
+{
+    afr_fd_ctx_t *fd_ctx = NULL;
+    afr_private_t *priv = NULL;
+    int i = 0;
+
+    priv = this->private;
+    fd_ctx = afr_fd_ctx_get(fd, this);
+    if (!fd_ctx)
+        return;
+
+    LOCK(&fd->lock);
+    {
+        for (i = 0; i < priv->child_count; i++) {
+            if (fd_ctx->opened_on[i] == AFR_FD_OPENING && priv->child_up[i] &&
```

Comment:
Yes, that could happen. Checking the local->need_open[] should suffice here, since that has the children info on which open is attempted in the first place. Removed this check.